docs/fastdds/dds_layer/topic/contentFilteredTopic/customFilters.rst

```diff
@@ -5,3 +5,109 @@
+------------------------------------------
+
+Fast DDS creates filters through a factory.
+Therefore a factory which provides instantiation of a Custom Filter must be implememented.
+
+A Custom Filter's factory has to inherites from |IContentFilterFactory-api|.
+This interface contains only two function to be implemented.
+
+Each time a Custom Filter has to be created or updated, |DomainParticipant-api| calls
+|IContentFilterFactory::create_content_filter-api| with next arguments.
+
+- ``filter_class_name`` - Filter class name for which the factory is being called.
+  Allows using the same factory for different filter classes.
+- ``type_name`` - Type name of the topic being filtered.
+- ``data_type`` - Type support object of the topic being filtered.
+- ``filter_expression`` - Custom filter expression.
+- ``filter_parameters`` - Values to set for the filter parameters (where custom filter expression has its pattern to
+  sustitutes them).
+- ``filter_instance`` - When a filter is being created, it will be nullptr on input, and will have the pointer to the
+  created filter instance on output.
+  When a filter is being updated, it will have a previously returned pointer on input.
+
+This function should return the results of the operation.
+
+
+When a Custom Filter should be removed, |DomainParticipant-api| calls
+|IContentFilterFactory::delete_content_filter-api|.
+The factory must remove the provided Custom Filter's instance.
+
+Next snippet code shows an example of Custom Filter's factory which manages instances of the Custom Filter implemented
+in previous section.
+
+.. literalinclude:: /../code/DDSCodeTester.cpp
+   :language: c++
+   :start-after: //DDS_CUSTOM_FILTER_FACTORY_CLASS
+   :end-before: //!
+   :dedent: 4
+
+.. _dds_layer_topic_customfilter_register:
+
+Registering the Factory
+-----------------------
+
+To be able to use the Custom Filter in an application, the Custom Filter's factory must be registered in the
+|DomainParticipant-api|. Next snippet code shows how to register a factory through API funciton
+|DomainParticipant::register_content_filter_factory-api|.
+
+.. literalinclude:: /../code/DDSCodeTester.cpp
+   :language: c++
+   :start-after: //DDS_CUSTOM_FILTER_REGISTER_FACTORY
+   :end-before: //!
+   :dedent: 8
+
+.. _dds_layer_topic_customfilter_create_topic:
+
+Creating a ContentFilteredTopic using the Custom Filter
+-------------------------------------------------------
+
+:ref:`dds_layer_topic_contentFilteredTopic_creation` explains how to create a |ContentFilteredTopic-api|.
+In the case of using a Custom Filter, |DomainParticipant::create_contentfilteredtopic-api| has an overload adding an
+argument to select the Custom Filter.
+
+Next snippet code shows how to create a |ContentFilteredTopic-api| using the Custom Filter.
+
+.. literalinclude:: /../code/DDSCodeTester.cpp
+   :language: c++
+   :start-after: //DDS_CUSTOM_FILTER_CREATE_TOPIC
+   :end-before: //!
+   :dedent: 8
```

Comment:
I suggest adding a note explaining that `DomainParticipant::delete_contentfilteredtopic` API is enough to delete Custom filter ContentFilteredTopic.

Reply:
This is already explained in the section about `ContentFilterTopic`.

Reply:
I do not completely agree. Section 3.5.8.1 explains how to create a ContentFilteredTopic and it is the section you are referring. However, how to delete a ContentFilteredTopic is explained in section 3.5.8.3. I do not mind adding another reference to this section, saying that to delete a custom ContentFilteredTopic please refer to this other section.